mr/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergRecordObjectInspector.java

```diff
@@ -35,21 +35,30 @@
 public final class IcebergRecordObjectInspector extends StructObjectInspector {
 
   private static final IcebergRecordObjectInspector EMPTY =
-          new IcebergRecordObjectInspector(Types.StructType.of(), Collections.emptyList());
+          new IcebergRecordObjectInspector(Types.StructType.of(), Collections.emptyList(), true);
 
   private final List<IcebergRecordStructField> structFields;
+  private final List<IcebergRecordStructField> structFieldsInLowercase;
```

Comment:
Why do we keep both of the fields? Shouldn't we just keep the currently requested version?

Reply:
You are right!